examples/seq2seq/seq2seq_trainer.py

```diff
@@ -0,0 +1,205 @@
+        return loss
+
+    def prediction_step(
+        self, model: nn.Module, inputs: Dict[str, Union[torch.Tensor, Any]], prediction_loss_only: bool
+    ) -> Tuple[Optional[float], Optional[torch.Tensor], Optional[torch.Tensor]]:
+        """
+        Perform an evaluation step on :obj:`model` using obj:`inputs`.
+
+        Subclass and override to inject custom behavior.
+
+        Args:
+            model (:obj:`nn.Module`):
+                The model to evaluate.
+            inputs (:obj:`Dict[str, Union[torch.Tensor, Any]]`):
+                The inputs and targets of the model.
+
+                The dictionary will be unpacked before being fed to the model. Most models expect the targets under the
+                argument :obj:`labels`. Check your model's documentation for all accepted arguments.
+            prediction_loss_only (:obj:`bool`):
+                Whether or not to return the loss only.
+
+        Return:
+            Tuple[Optional[float], Optional[torch.Tensor], Optional[torch.Tensor]]:
+            A tuple with the loss, logits and labels (each being optional).
+        """
+        has_labels = any(inputs.get(k) is not None for k in ["labels", "lm_labels", "masked_lm_labels"])
+
+        inputs = self._prepare_inputs(inputs)
+
+        max_length = (
+            model.config.max_generate_length
+            if hasattr(model.config, "max_generate_length")
+            else model.config.max_position_embeddings
+        )
+
+        with torch.no_grad():
+            if self.args.predict_with_generate:
+                generated_tokens = model.generate(
+                    inputs["input_ids"],
+                    attention_mask=inputs["attention_mask"],
+                    use_cache=True,
+                    num_beams=model.config.num_beams,
+                    max_length=max_length,
+                )
+                # in case the batch is shorter than max length, the output should be padded
+                generated_tokens = self._pad_tensors_to_max_len(
+                    generated_tokens, max_length, model.config.pad_token_id
+                )
+
+                if has_labels:
+                    outputs = model(**inputs)
+                    loss = outputs[0]
+                    loss = loss.mean().item()
+                else:
+                    loss = None
+            else:
+                outputs = model(**inputs)
+                if has_labels:
+                    loss, logits = outputs[:2]
+                    loss = loss.mean().item()
+                else:
+                    loss = None
+                    logits = outputs[0]
+
+        if prediction_loss_only:
+            return (loss, None, None)
+
+        labels_out = inputs.get("labels")
+        if labels_out is not None:
+            labels_out = labels_out.detach()
+            labels = self._pad_tensors_to_max_len(labels_out, max_length, model.config.pad_token_id)
+        return (loss, logits.detach(), labels)
```

Comment:
Where is logits initialized if predict_with_generate is True?

Reply:
should it perhaps be set to generated_tokens in line 172?

Reply:
Thanks for catching this @melody-ju :) Yes, `logits` was renamed to `generated_tokens` in a recent commit, to make naming clear. Need to use return  `generated_tokens` when `predict_with_generate` is `True` otherwise `logits`